osu.Game/Graphics/UserInterface/Counter.cs

```diff
@@ -0,0 +1,110 @@
+﻿//Copyright (c) 2007-2016 ppy Pty Ltd <contact@ppy.sh>.
+//Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using OpenTK;
+using OpenTK.Graphics;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+
+namespace osu.Game.Graphics.UserInterface
+{
+    public abstract class Count : AutoSizeContainer
+    {
+        private Sprite buttonSprite;
+        private Sprite glowSprite;
+        private SpriteText keySpriteText;
+        private SpriteText countSpriteText;
+
+        public override string Name { get; }
+        public KeyCounter ParentCounter { get; set; }
+        public int Counts { get; private set; }

```

Comment:
Not sure why this is plural. If it's due to similarity with the class name, i think we should probably rename all the classes a bit...

````
KeyCounterCollection
    - KeyCounterMouse : KeyCounter
    - KeyCounterMouse : KeyCounter
    - KeyCounterKeyboard : KeyCounter
    - KeyCounterKeyboard : KeyCounter
````